charms/kserve-controller/src/charm.py

```diff
@@ -418,11 +434,14 @@ def send_object_storage_manifests(self):
             "secret_name": f"{self.app.name}-s3",
         }
 
-        self._send_manifests(interfaces, secrets_context, SECRETS_FILES, "secrets")
-        self._send_manifests(
-            interfaces, service_accounts_context, SERVICE_ACCOUNTS_FILES, "service-accounts"
+        secrets_manifests = self._create_manifests(SECRETS_FILES, secrets_context)
+        service_accounts_manifests = self._create_manifests(
+            SERVICE_ACCOUNTS_FILES, service_accounts_context
         )
 
+        self.secrets_manifests_wrapper.send_data(secrets_manifests)
+        self.service_accounts_manifests_wrapper.send_data(service_accounts_manifests)
+
```

Comment:
**nit:** WDYT about keeping the send_manifests like you did in [mlflow](https://github.com/canonical/mlflow-operator/pull/220/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR405-R410)? Since we do this more than once, I think it would make sense.